### PR TITLE
CORE-10180: Escape string parameters when creating json to prevent possible injection attacks

### PR DESCRIPTION
**Pull request by anien**
Changes:
* Escape string parameters
* Fix most of the kdoc parameters
* Add tests

### CODE REVIEW

File: tools/plugins/initial-config/src/test/kotlin/net/corda/cli/plugin/initialconfig/TestInitialConfigPluginDb.kt [hunk 1]
```diff
@@ -130,4 +132,54 @@ class TestInitialConfigPluginDb {
...
+        val password = ""
+        val salt = "123"
+        val passphrase = "123"
+        val vaultKey = "corda-config-database-password"
+        val secretsService: SecretsCreateService = EncryptionSecretsServiceImpl(passphrase, salt)
+
+        val outText = createConfigDbConfig(jdbcUrl, username, password, vaultKey, jdbcPoolMaxSize, jdbcPoolMinSize,
+            idleTimeout, maxLifetime, keepaliveTime, validationTimeout, secretsService)
+
+        println(outText)
```
Comment (dickon) [inline tools/plugins/initial-config/src/test/kotlin/net/corda/cli/plugin/initialconfig/TestInitialConfigPluginDb.kt]: debug code to delete?

**Review by dickon — DISMISSED** [review]
There's a println that should get removed, and it feels like a bit of extra test coverage would be worth it.

Also I wish we were using a function in a library we already depend on for this, though I don't know of one.

**Review by ronanbrowne — DISMISSED** [review]
LGTM (build file only reviewed)

File: tools/plugins/initial-config/src/test/kotlin/net/corda/cli/plugin/initialconfig/TestInitialConfigPluginDb.kt (hunk 1 above)
Comment (anien) [reply tools/plugins/initial-config/src/test/kotlin/net/corda/cli/plugin/initialconfig/TestInitialConfigPluginDb.kt]: Done

**Comment by anien** [comment]
> There's a println that should get removed, and it feels like a bit of extra test coverage would be worth it.
> 
> Also I wish we were using a function in a library we already depend on for this, though I don't know of one.

Println removed. I am happy to talk about the test coverage. As there is a difference of the checks/escaping depending on how the code is called, I covered both scenarios in the new tests. Not sure what else to add.

**Review by dickon — APPROVED** [review]
LGTM, same comments as before

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Pull request merged**